$NetBSD: patch-aa,v 1.3 2010/09/29 11:21:02 obache Exp $

* some platforms don't support LOG_AUTHPRIV and LOG_FTP syslog facility
* remove deprecated `errno` definition

--- src/nrpe.c.orig	2008-03-10 21:04:43.000000000 +0000
+++ src/nrpe.c
@@ -613,10 +613,14 @@ int get_log_facility(char *varvalue){
 		log_facility=LOG_UUCP;
 	else if(!strcmp(varvalue,"cron"))
 		log_facility=LOG_CRON;
+#ifdef LOG_AUTHPRIV
 	else if(!strcmp(varvalue,"authpriv"))
 		log_facility=LOG_AUTHPRIV;
+#endif
+#ifdef LOG_FTP
 	else if(!strcmp(varvalue,"ftp"))
 		log_facility=LOG_FTP;
+#endif
 	else if(!strcmp(varvalue,"local0"))
 		log_facility=LOG_LOCAL0;
 	else if(!strcmp(varvalue,"local1"))
@@ -1320,7 +1324,6 @@ int my_system(char *command,int timeout,
         pid_t pid;
 	int status;
 	int result;
-	extern int errno;
 	char buffer[MAX_INPUT_BUFFER];
 	int fd[2];
 	FILE *fp;
